driver/network_driver.go

```diff
@@ -260,68 +298,89 @@ func (d NetworkDriver) CreateEndpoint(request *network.CreateEndpointRequest) (*
 		addresses = append(addresses, caliconet.IPNet{IPNet: *ipnet})
 	}
 
+	wepName, err := d.generateEndpointName(hostname, request.EndpointID)
+	if err != nil {
+		log.Errorln(err)
+		return nil, err
+	}
+
 	endpoint := api.NewWorkloadEndpoint()
-	endpoint.Metadata.Node = hostname
-	endpoint.Metadata.Orchestrator = d.orchestratorID
-	endpoint.Metadata.Workload = d.containerName
-	endpoint.Metadata.Name = request.EndpointID
+	endpoint.Name = wepName
+	//endpoint.ObjectMeta.Namespace = fmt.Sprintf("%s.%s", d.orchestratorID, networkName)
+	endpoint.ObjectMeta.Namespace = d.namespace
```

Comment:
I think it will be a bit hard for getting network_name when join/delete a endpoint. Docker call those interfaces with ID, we have to query docker daemon. In my opinion cyclic call is not a good idea. So let's keep namespace as libnetwork.

Reply:
sgtm